beacon-chain/rpc/beacon_chain_server.go

```diff
@@ -196,14 +175,125 @@ func (bs *BeaconChainServer) GetValidatorQueue(
 	return nil, status.Error(codes.Unimplemented, "not implemented")
 }
 
-// ListValidatorAssignments retrieves the validator assignments for a given epoch.
+// ListValidatorAssignments retrieves the validator assignments for a given epoch,
+// optional validator indices or public keys may be included to filter validator assignments.
 //
-// This request may specify optional validator indices or public keys to
-// filter validator assignments.
+// TODO(#3045): Implement validator set for a specific epoch. Current implementation returns latest set,
+// this is blocked by DB refactor.
 func (bs *BeaconChainServer) ListValidatorAssignments(
 	ctx context.Context, req *ethpb.ListValidatorAssignmentsRequest,
 ) (*ethpb.ValidatorAssignments, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	if int(req.PageSize) > params.BeaconConfig().MaxPageSize {
```

Comment:
could you break it into simpler functions ? maybe handling the request with a filter as one function 
and providing the default response as another